x/btcstaking/types/pop.go

```diff
@@ -41,14 +41,20 @@ func (pop *ProofOfPossessionBTC) ToHexStr() (string, error) {
 }
 
 // Verify that the BTC private key corresponding to the bip340PK signed the staker address
-func (pop *ProofOfPossessionBTC) Verify(staker sdk.AccAddress, bip340PK *bbn.BIP340PubKey, net *chaincfg.Params) error {
+func (pop *ProofOfPossessionBTC) Verify(
+	signingContext string,
+	address sdk.AccAddress,
+	bip340PK *bbn.BIP340PubKey,
+	net *chaincfg.Params,
+) error {
+	stakerBech32Addr := address.String()
 	switch pop.BtcSigType {
 	case BTCSigType_BIP340:
-		return pop.VerifyBIP340(staker, bip340PK)
+		return pop.VerifyBIP340(signingContext, address, bip340PK)
 	case BTCSigType_BIP322:
-		return pop.VerifyBIP322(staker, bip340PK, net)
+		return pop.VerifyBIP322(signingContext, address, bip340PK, net)
 	case BTCSigType_ECDSA:
-		return pop.VerifyECDSA(staker, bip340PK)
+		return pop.VerifyECDSA(signingContext+stakerBech32Addr, bip340PK)
```

Comment:
suggestion for us to keep the pattern as the signingContext as a first parameter